#!/usr/bin/env python
PACKAGE = 'rr100_drive_amp'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('active', bool_t, 0, 'Whether the drive amp is active or not', True)

gen.add('p_gain', double_t, 0, 'The proportional gain to apply to received velocity commands', 1.0, 0.0, 100.0)
gen.add('i_gain', double_t, 0, 'The integral gain to apply to received velocity commands',     0.0, 0.0, 20.0)
gen.add('d_gain', double_t, 0, 'The derivational gain to apply to received velocity commands', 0.0, 0.0, 5.0)

exit(gen.generate(PACKAGE, "pid_reconf_node", 'DriveAmp'))